File: lecture_003(Pattern)/l_002_pattern2.java
import java.util.Scanner;
public class l_002_pattern2 {
	public static Scanner scn = new Scanner(System.in);
	public static void main(String[] args) {
		int n = scn.nextInt();
		int nst = 1;
		int nsp = n-1;
		patternMirror(n,nst,nsp);
	}
	
	public static void patternMirror(int n, int nst, int nsp) {
		for(int r=1; r<=n; r++) {
			for(int cst=1; cst<=nst; cst++) {
				System.out.print("*");
			}
			for(int csp=1; csp<=nsp; csp++) {
				System.out.print(" ");
			}
			nst++;
			nsp--;
			System.out.println();
		}
	}
	
}

File: lecture_003(Pattern)/l_002_pattern2.cpp
#include<iostream>
using namespace std;

void patternmirror(int n) {
    int nst=1;
    int nsp=n-1;
    for(int r=1; r<=n; r++) {
        for(int cst=1; cst<=nst; cst++) {
            cout<<"*";
        }
        for(int csp=1; csp<=nsp; csp++) {
            cout<<"-";
        }
        nst++;
        nsp--;
        cout<<endl;
    }
} 
int main(int args, char**argvs) {
    int n;
    cin>>n;
    patternmirror(n);
    return 0;
}